tests/mlir/pocl-opt/affine_parallel_to_for_test.mlir

```diff
@@ -0,0 +1,57 @@
+// RUN: pocl-opt --pocl-affine-parallel-to-for --allow-unregistered-dialect --split-input-file %s | FileCheck %s
```

Comment:
These tests look quite "fragile", I'm awaiting challenges when upgrading LLVMs. Let's hope I'm wrong.

Reply:
I made this affine paralel to for test more robust, by using the 'dummy' test-dialect, which seems to be commonly used in MLIR tests, since the loop body ops do no matter in the test.

For the workgroup-test I wasn't able to simplify much, since it is actually testing kind of the core functionality of the Workgroup.cc-pass. So that one is likely to be useful when fixing upgrade issues.